### PR TITLE
Add LayoutLM processor

### PR DESCRIPTION
**Pull request by gau-nernst**
# What does this PR do?

Fixes #27826

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker and @younesbelkada

### CODE REVIEW

**Comment by gau-nernst** [comment]
The tests are failing because when I copy LayoutLMv2Tokenizer to LayoutLMTokenizer, `bbox` input is required. To preserve backward-compatibility, we need to check if `bbox` is None, and apply the old logic in that case. I'm not familiar with the tokenizer code, so help would be appreciated 🙏.

**Comment by gau-nernst** [comment]
The tests are failing because when I copy LayoutLMv2Tokenizer to LayoutLMTokenizer, `bbox` input is required. To preserve backward-compatibility, we need to check if `bbox` is None, and apply the old logic in that case. I'm not familiar with the tokenizer code, so help would be appreciated 🙏.

File: src/transformers/models/layoutlm/tokenization_layoutlm.py [hunk 1]
```diff
@@ -210,7 +320,7 @@ def build_inputs_with_special_tokens(
     ) -> List[int]:
         """
         Build model inputs from a sequence or a pair of sequence for sequence classification tasks by concatenating and
-        adding special tokens. A LayoutLM sequence has the following format:
+        adding special tokens. A BERT sequence has the following format:
```
Comment (ArthurZucker) [inline src/transformers/models/layoutlm/tokenization_layoutlm.py:323]: that also looks strange

File: src/transformers/models/layoutlm/tokenization_layoutlm.py [hunk 2]
```diff
@@ -307,6 +411,926 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
...
+        tokens = []
+        pair_tokens = []
+        token_boxes = []
+        pair_token_boxes = []
+        labels = []
+
+        if text_pair is None:
+            if word_labels is None:
+                # CASE 1: document image classification (training + inference) + CASE 2: token classification (inference)
+                for word, box in zip(text, boxes):
```
Comment (ArthurZucker) [inline src/transformers/models/layoutlm/tokenization_layoutlm.py:976]: this seems to be the source of the issue. You need to make sure there are any boxes before potential doing anything else. The tests rely on this and they are a good indication of how to keep backward compatibility !

**Review by ArthurZucker — COMMENTED** [review]
Part of the test should be updated, but otherwise it will help us make sure we are BC!

File: src/transformers/models/layoutlm/processing_layoutlm.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            FutureWarning,
+        )
+        return self.image_processor_class
+
+    @property
+    def feature_extractor(self):
+        warnings.warn(
+            "`feature_extractor` is deprecated and will be removed in v5. Use `image_processor` instead.",
+            FutureWarning,
+        )
```
[the remark this answers is not in the document]
Comment (gau-nernst) [reply src/transformers/models/layoutlm/processing_layoutlm.py:184]: ok

File: src/transformers/models/layoutlm/tokenization_layoutlm.py [hunk 4]
```diff
@@ -262,15 +372,9 @@ def create_token_type_ids_from_sequences(
...
-
-        ```
-        0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1
-        | first sequence    | second sequence |
-        ```
-
-        If `token_ids_1` is `None`, this method only returns the first portion of the mask (0s).
+        Create a mask from the two sequences passed to be used in a sequence-pair classification task. A BERT sequence
+        pair mask has the following format: :: 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 | first sequence | second
+        sequence | If `token_ids_1` is `None`, this method only returns the first portion of the mask (0s).
```
[the remark this answers is not in the document]
Comment (gau-nernst) [reply src/transformers/models/layoutlm/tokenization_layoutlm.py:377]: I use the code cloning to get this (`# Copied from transformers.models.layoutlmv2.tokenization_layoutlmv2.LayoutLMv2Tokenizer with LayoutLMv2->LayoutLM,LAYOUTLMV2->LAYOUTLM`). Seems like there are some issues with the code cloning. I will double-check and fix it.

**Comment by ArthurZucker** [comment]
(feel free to ping me again for another review 😉

**Pull request closed without merging**